### PR TITLE
Front: Enable description and logo for methods in checkout (SHOOP-2406)

### PR DESCRIPTION
**Pull request by tulimaki**
* Create custom method widget to render method logo and description in
checkout.
* Group methods by service provider and also show service provider
logo or name at the methods phase in checkout

### CODE REVIEW

File: shoop/front/checkout/methods.py [hunk 1]
```diff
@@ -8,52 +8,65 @@
...
+        grouped_methods[getattr(method, method.provider_attr)].append(method)
+    return grouped_methods
+
+
 class MethodModelChoiceField(forms.ModelChoiceField):
     def __init__(self, *args, **kwargs):
-        self.request = kwargs.pop("request", None)
-        self.basket = kwargs.pop("basket", None)
-        self.show_prices = bool(kwargs.pop("show_prices", True))
         super(MethodModelChoiceField, self).__init__(*args, **kwargs)
```
Comment (shawnadelic) [inline shoop/front/checkout/methods.py]: Unnecessary now?

File: shoop/themes/classic_gray/templates/classic_gray/shoop/front/checkout/method_choice.jinja [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<div id="id_{{ field_name }}">
+{% for service_provider, services in grouped_methods.items() %}
+    <div class="service-provider">
+        {% if service_provider.logo %}
+            <img class="image" src="{{ service_provider.logo.url }}" alt="{{ service_provider.name }}">
+         {% else %}
```
Comment (juhakujala) [inline shoop/themes/classic_gray/templates/classic_gray/shoop/front/checkout/method_choice.jinja]: Indentation off here

File: shoop/themes/classic_gray/templates/classic_gray/shoop/front/checkout/method_choice.jinja [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<div id="id_{{ field_name }}">
+{% for service_provider, services in grouped_methods.items() %}
+    <div class="service-provider">
+        {% if service_provider.logo %}
+            <img class="image" src="{{ service_provider.logo.url }}" alt="{{ service_provider.name }}">
```
Comment (juhakujala) [inline shoop/themes/classic_gray/templates/classic_gray/shoop/front/checkout/method_choice.jinja]: should we crop these to max 250px or something?

File: shoop/themes/classic_gray/templates/classic_gray/shoop/front/checkout/method_choice.jinja [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    <div class="radio method-radio">
+    {% for service in services %}
+        <label for="id_{{ field_name }}_{{ service.id }}">
+            <input
+                {% if current_value == service.id %}checked=checked{% endif %}
+                name="{{ field_name }}"
+                id="id_{{ field_name }}_{{ service.id }}"
+                type="radio" value="{{ service.id }}">
+            {% if service.logo %}
+            <img class="image" src="{{ service.logo.url }}" alt="{{ service.name }}">
```
Comment (juhakujala) [inline shoop/themes/classic_gray/templates/classic_gray/shoop/front/checkout/method_choice.jinja]: Same cropping here

File: shoop/front/checkout/methods.py (hunk 1 above)
Comment (tulimaki) [inline shoop/front/checkout/methods.py]: True, I think we don't even need that MethdoModelChoiceField at all...

**Pull request merged**